src/containers/TriggerBinding/TriggerBinding.js

```diff
@@ -0,0 +1,308 @@
+  /* eslint-disable react/sort-comp */
+  state = initialState;
+
+  static notification({ kind, message, intl }) {
+    const titles = {
+      info: intl.formatMessage({
+        id: 'dashboard.triggerBinding.unavailable',
+        defaultMessage: 'TriggerBinding not available'
+      }),
+      error: intl.formatMessage({
+        id: 'dashboard.triggerBinding.errorloading',
+        defaultMessage: 'Error loading TriggerBinding'
+      })
+    };
+    return (
+      <InlineNotification
+        kind={kind}
+        hideCloseButton
+        lowContrast
+        title={titles[kind]}
+        subtitle={message}
+      />
+    );
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { match, webSocketConnected } = this.props;
+    const { namespace, triggerBindingName } = match.params;
+    const {
+      match: prevMatch,
+      webSocketConnected: prevWebSocketConnected
+    } = prevProps;
+    const {
+      namespace: prevNamespace,
+      triggerBindingName: prevTriggerBindingName
+    } = prevMatch.params;
+
+    if (
+      namespace !== prevNamespace ||
+      triggerBindingName !== prevTriggerBindingName ||
+      (webSocketConnected && prevWebSocketConnected === false)
+    ) {
+      this.fetchData();
+    }
+  }
+
+  fetchData() {
+    const { match } = this.props;
+    const { namespace, triggerBindingName } = match.params;
+    this.props.fetchTriggerBinding({ name: triggerBindingName, namespace });
+  }
+
+  handleTabSelect = index => {
+    switch (index) {
+      case 0:
+        this.setState({
+          showYAML: false,
+          showDetails: true
+        });
+        break;
+
+      case 1:
+        this.setState({
+          showYAML: true,
+          showDetails: false
+        });
+        break;
+
+      default:
+        this.setState({
+          showYAML: false,
+          showDetails: false
+        });
+    }
+  };
+
+  render() {
+    const {
+      intl,
+      error,
+      match,
+      selectedNamespace,
+      triggerBinding
+    } = this.props;
+    const { showYAML, showDetails } = this.state;
+    const { triggerBindingName } = match.params;
+
+    if (error) {
+      return TriggerBindingContainer.notification({
+        kind: 'error',
+        intl
+      });
+    }
+    if (!triggerBinding) {
+      return TriggerBindingContainer.notification({
+        kind: 'error',
+        message: 'TriggerBinding not available',
+        intl
+      });
+    }
+
+    let labels = '';
+    const formattedLabelsToRender = [];
+    if (triggerBinding.metadata.labels) {
+      labels = JSON.stringify(triggerBinding.metadata.labels)
+        .replace('{', '')
+        .replace('}', '')
+        .replace(/['"]+/g, '')
+        .replace('$', '');
+
+      const labelsSplitOnComma = labels.split(',');
+      labelsSplitOnComma.forEach(foo => {
+        const splitOnColon = foo.split(':');
+        formattedLabelsToRender.push(`  ${splitOnColon[0]}: ${splitOnColon[1]}
```

Comment:
why do we add leading whitespace?
can we rename `foo` to something more meaningful?

also, for readability you can do:

```js
const [key, value] = label.split(':');
...push(`${key}: ${value}`);
```